lib/ansible/plugins/lookup/cyberarkpassword.py

```diff
@@ -0,0 +1,129 @@
+import os
+import subprocess
+
+from ansible.errors import AnsibleError
+from ansible.plugins.lookup import LookupBase
+from ansible.module_utils._text import to_text
+
+CLIPASSWORDSDK_CMD = os.getenv('AIM_CLIPASSWORDSDK_CMD', '/opt/CARKaim/sdk/clipasswordsdk')
+
+
+class CyberarkPassword:
+
+    def __init__(self, appid=None, query=None, output=None, **kwargs):
+
+        self.appid = appid
+        self.query = query
+        self.output = output
+
+        # Support for Generic parameters to be able to specify
+        # FailRequestOnPasswordChange, Queryformat, Reason, etc.
+        self.extra_parms = []
+        for key, value in kwargs.items():
+            self.extra_parms.append('-p')
+            self.extra_parms.append("%s=%s" % (key, value))
+
+        if self.appid is None:
+            raise AnsibleError("CyberArk Error: No Application ID specified")
+        if self.query is None:
+            raise AnsibleError("CyberArk Error: No Vault query specified")
+
+        if self.output is None:
+            # If no output is specified, return at least the password
+            self.output = "password"
+        else:
+            # To avoid reference issues/confusion to values, all
+            # output 'keys' will be in lowercase.
+            self.output = self.output.lower()
+
+        self.delimiter = "@#@"  # Known delimiter to split output results
+
+    def get(self):
+
+        result_dict = {}
+
+        try:
+            allParms = [
+                CLIPASSWORDSDK_CMD,
+                'GetPassword',
+                '-p', 'AppDescs.AppID=%s' % self.appid,
+                '-p', 'Query=%s' % self.query,
+                '-o', self.output,
+                '-d', self.delimiter]
+            allParms.extend(self.extra_parms)
+
+            credential = subprocess.check_output(allParms)
+
+            if credential and credential.endswith(b'\n'):
+                credential = credential[:-1]
+
+            output_names = self.output.split(",")
+            output_values = credential.split(self.delimiter)
+
+            for i in range(len(output_names)):
+                if output_names[i].startswith("passprops."):
+                    if "passprops" not in result_dict:
+                        result_dict["passprops"] = {}
+                    output_prop_name = output_names[i][10:]
+                    result_dict["passprops"][output_prop_name] = output_values[i]
+                else:
+                    result_dict[output_names[i]] = output_values[i]
+
+        except subprocess.CalledProcessError as e:
+            raise AnsibleError(e.output)
+        except OSError as e:
+            raise AnsibleError("ERROR - AIM not installed or clipasswordsdk not in standard location. ERROR=(%s) => %s " % (to_text(e.errno), e.strerror))
+
+        return [result_dict]
+
+
+class LookupModule(LookupBase):
+
+    """
+    USAGE: {{lookup('cyberarkpassword', appid='Application1', query='safe=Safe1;Folder=root;Object=User1',
```

Comment:
Note that using keyword arguments for lookup parameters will prevent to use the `with_cyberarkpassword` form (because [it's not possible to set arguments](https://github.com/cyberark-bizdev/ansible/blob/0c39bbaa6974e5d3c54ba19b13f7a814619d0ce3/lib/ansible/executor/task_executor.py#L215) when the `with_cyberarkpassword` form is used).

I mean using the `cyberarkpassword` lookup like that isn't possible:
```
- debug:
    msg: 'pass: {{ item }}'
  with_password:
    - /home/pilou/mysecret length=32
```

 The alternative to keyword arguments is to pass parameters inside `term`, here are examples: [password](https://github.com/cyberark-bizdev/ansible/blob/0c39bbaa6974e5d3c54ba19b13f7a814619d0ce3/lib/ansible/plugins/lookup/password.py#L223), [hashi_vault](https://github.com/cyberark-bizdev/ansible/blob/0c39bbaa6974e5d3c54ba19b13f7a814619d0ce3/lib/ansible/plugins/lookup/hashi_vault.py#L135), [keyring](https://github.com/cyberark-bizdev/ansible/blob/0c39bbaa6974e5d3c54ba19b13f7a814619d0ce3/lib/ansible/plugins/lookup/keyring.py#L58).

All lookup plugins allow to use the `with_lookup` form ([see last paragraph here](http://docs.ansible.com/ansible/playbooks_loops.html#standard-loops): `any lookup plugin can be used as a source for a loop`) - even if some options of some lookup plugins [aren't settable](https://github.com/cyberark-bizdev/ansible/blob/0c39bbaa6974e5d3c54ba19b13f7a814619d0ce3/lib/ansible/plugins/lookup/credstash.py#L32) when used as a source for loop, they are still usable - the current implementation of the cyberarkpassword lookup plugin prevent that.

Not sure if the implemented behavior is a problem but it should be at least documented (something like: `cyberarkpassword lookup can not be used as a source for loop`).